### PR TITLE
Removed boost/string methods in favour of utility methods in util/Name.h

### PR DESCRIPTION
**Pull request by Idclip**
Signed-off-by: Nick Avramoussis <4256455+Idclip@users.noreply.github.com>

### CODE REVIEW

File: openvdb/io/Compression.cc [hunk 1]
```diff
@@ -22,11 +21,12 @@ compressionToString(uint32_t flags)
...
-    std::vector<std::string> words;
-    if (flags & COMPRESS_ZIP) words.push_back("zip");
-    if (flags & COMPRESS_BLOSC) words.push_back("blosc");
-    if (flags & COMPRESS_ACTIVE_MASK) words.push_back("active values");
-    return boost::join(words, " + ");
+    std::string descr;
+    if (flags & COMPRESS_ZIP) descr += "zip + ";
+    if (flags & COMPRESS_BLOSC) descr += "blosc + ";
+    if (flags & COMPRESS_ACTIVE_MASK) descr += "active values + ";
+    if (!descr.empty()) descr.resize(descr.size() - 3);
```
Comment (danrbailey) [inline openvdb/io/Compression.cc:28]: This is pretty clunky. You can't have ZIP and BLOSC defined at the same time, so I think I'd prefer this to just be explicit for readability:

```
if (flags & COMPRESS_ZIP) {
    if (flags & COMPRESS_ACTIVE_MASK)    return "zip + active values";
    else                                 return "zip";
}
// etc
```

File: openvdb/util/Name.h [hunk 2]
```diff
@@ -35,6 +37,77 @@ writeString(std::ostream& os, const Name& name)
     os.write(&name[0], size);
 }
 
+namespace util {
```
Comment (danrbailey) [inline openvdb/util/Name.h:40]: I think I'd be tempted to put these in a new namespace for extra readability - openvdb::util::split() doesn't tell you anything about what it is you're actually splitting for example. Perhaps openvdb::string::split() would be clearer?

There are other non-util namespaces defined in this util sub-directory (such as openvdb::logging) so it wouldn't be breaking with precedence there.

File: openvdb/util/Name.h [hunk 3]
```diff
@@ -35,6 +37,77 @@ writeString(std::ostream& os, const Name& name)
...
+{
+    ltrim(s);
+    rtrim(s);
+}
+
+inline void to_lower(std::string& s)
+{
+    std::transform(s.begin(), s.end(), s.begin(),
+        [](auto c){ return std::tolower(c); });
+}
```
Comment (danrbailey) [inline openvdb/util/Name.h:107]: Would be worth adding unit tests for these new methods.

File: openvdb/util/Name.h (hunk 2 above)
Comment (Idclip) [reply openvdb/util/Name.h:40]: Happy to go with `openvdb::string` for now

File: openvdb/io/Compression.cc (hunk 1 above)
Comment (Idclip) [reply openvdb/io/Compression.cc:28]: Technically, as far as this function is concerned, blosc and zip flags can exist at the same time as they're different bit toggles i.e. compressionToString(7) == "zip + blosc + active values". I realise that it shouldn't be (isn't) possible for them to both to exist, but seems that this method works without that context

Comment (danrbailey) [reply openvdb/io/Compression.cc:28]: Perhaps we error or assert that both are not supplied at once, I still prefer the explicit solution here for readability.

**Comment by danrbailey** [comment]
Looks good to me. If you just make the switch from openvdb::util to openvdb::string and maybe add a line of two of comments to each of these new methods as way of documentation, I'm happy to approve this.

**Comment by danrbailey** [comment]
@Idclip - can this one be merged now?

**Pull request merged**